src/main/database/driveFoldersDB.ts

```diff
@@ -0,0 +1,110 @@
+import { DatabaseAdapter } from './adapters/base';
+import { RemoteSyncedFolder } from '../remote-sync/helpers';
+import SQLite from 'better-sqlite3';
+
+export function generateCreateTableQuery(): string {
+  const query = `CREATE TABLE IF NOT EXISTS drive_folders (
+    type TEXT NOT NULL,
+    id INTEGER UNIQUE NOT NULL,
+    parentId INTEGER,
+    bucket TEXT,
+    userId INTEGER, 
+    createdAt TEXT,
+    updatedAt TEXT NOT NULL,
+    uuid TEXT NOT NULL,
+    plainName TEXT,
+    status TEXT NOT NULL,
+    PRIMARY KEY (uuid)
+  )`;
+
+  return query;
+}
+
+export class DriveFoldersDB implements DatabaseAdapter<RemoteSyncedFolder> {
+  private _db?: SQLite.Database;
+  async connect(): Promise<{ success: boolean }> {
+    this._db = SQLite('drive_synced.db', {});
+
+    this.db.prepare(generateCreateTableQuery()).run();
+    return {
+      success: this._db.open,
+    };
+  }
+
+  async get(folderUuid: string) {
```

Comment:
Could we use direct references to the entities as types?

So, instead of:
`async get(folderUuid: string) {`

We can do:
`async get(folderUuid: RemoteSyncedFolder['uuid']) {`

Also, as this is the folders database, I find it a bit verbose to say `folderThis` `folderThat` to the namings. So, as you have called the methods get instead of getFolder, I would suggest calling params as `uuid` instead of `folderUuid`, just adding the entity before if it is a foreign entity, like `fileUuid`, etc